"Fatal: can't create directory '/github/home/.gnupg': File exists" bug should be fixed

sometimes I got the following similar errors:

```bash
##[error]Error: Command failed: cat /tmp/tmp-7-KO9Z5nBzpauj | gpg --quiet --batch --yes --decrypt --passphrase-fd=0 /tmp/tmp-7-xWWPkjYsReqI
gpg: Fatal: can't create directory '/github/home/.gnupg': File exists

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -106,7 +106,6 @@ jobs:
 
     - name: Docker push
       run: |
-        cat $DOCKER_CONFIG/config.json
         docker tag teracyhq-incubator/secret-manager-action:develop $CI_REGISTRY_IMAGE/secret-manager-action:$IMG_TAG
         echo "::debug::docker push $CI_REGISTRY_IMAGE/secret-manager-action:$IMG_TAG"
         docker push $CI_REGISTRY_IMAGE/secret-manager-action:$IMG_TAG

--- a/src/gpg.ts
+++ b/src/gpg.ts
@@ -2,7 +2,7 @@ import * as util from "util";
 import { promises as fs } from "fs";
 
 import * as io from "@actions/io";
-import { file } from "tmp-promise";
+import { file, dir } from "tmp-promise";
 
 // check if ggp is available, othewise, throw error
 (async function (): Promise<void> {
@@ -29,6 +29,10 @@ export async function decrypt(
 ): Promise<string> {
   const { path: passphrasePath, cleanup: cleanupPassPhrase } = await file();
   const { path: filePath, cleanup: cleanupFile } = await file();
+  // see: https://github.com/teracyhq-incubator/secret-manager-action/issues/17
+  const { path: gpgHomeDir, cleanup: cleanupGpgDir } = await dir({
+    unsafeCleanup: true,
+  });
 
   await fs.writeFile(passphrasePath, opts.passphrase);
   await fs.writeFile(filePath, input);
@@ -36,7 +40,7 @@ export async function decrypt(
   const execCmd = [
     `cat ${passphrasePath}`,
     "|",
-    `gpg --quiet --batch --yes --decrypt --passphrase-fd=0 ${filePath}`,
+    `gpg --homedir=${gpgHomeDir} --quiet --batch --yes --decrypt --passphrase-fd=0 ${filePath}`,
   ].join(" ");
 
   const { stdout, stderr } = await exec(execCmd);
@@ -47,5 +51,6 @@ export async function decrypt(
 
   cleanupPassPhrase();
   cleanupFile();
+  cleanupGpgDir();
   return stdout;
 }

